custom_components/weenect/device_tracker.py

```diff
@@ -140,7 +140,11 @@ def __init__(
     @property
     def available(self) -> bool:
         """Return if entity is available."""
-        return super().available and bool(self.coordinator.data[self.id]["position"])
+        return (
```

Comment:
I concur for all others but am hesitant for this one. I would like to show the device tracker on the last known location on the map. Does this still work with this change?

Reply:
Yes but only if there are available states within the `hours_to_show` window. TBH the device_tracker was the sensor that initiated my train of thought for changing the availability logic. I wasn't too happy being mislead that the tracker is in zone X when it is actually off or has no signal and also looking at the history it is IMHO better to see unavailability when offline.

Reply:
Very good point!

Reply:
We could also go for a different logic by not updating the attributes of the tracker when `valid_signal` is `false`. Currently if the API provides lat/lng, HA will update no matter if the data is 20 hours old.

Reply:
No, you have me completely convinced with your initial proposal